src/textual/document/_languages.py

```diff
@@ -1,13 +1,43 @@
 BUILTIN_LANGUAGES = sorted(
     [
-        "markdown",
-        "yaml",
-        "sql",
+        "bash",
+        "c",
+        "c_sharp",
+        "cpp",
         "css",
+        "dockerfile",
+        "dot",
+        "elisp",
+        "elixir",
+        "elm",
+        "embedded_template",
+        "erlang",
+        "fortran",
+        "go",
+        "gomod",
+        "hack",
+        "haskell",
         "html",
+        "java",
+        "javascript",
+        "jsdoc",
         "json",
+        "kotlin",
+        "make",
+        "markdown",
+        "objc",
+        "ocaml",
+        "php",
         "python",
+        "r",
         "regex",
+        "ruby",
+        "rust",
+        "scala",
+        "sql",
+        "sqlite",
         "toml",
+        "typescript",
```

Comment:
Typescript has issues
d
```
NameError: Invalid node type enum_definition
```

Reply:
Yeah I've found that this means the parser and highlights file is out of sync: the highlight query file has been updated to refer to a node type that the parser is unaware of. Usually moving to an older version of the highlights file (the version that "aligns" with the parser) fixes the problem.

Reply:
Ahh, you're right. Once I pin on `tree-sitter-languages==1.10.2`, which is the version I used to map to the `.scm` files, everything works.

Reply:
typescript highlighting is non existent though after the new version pinning
<img width="1840" alt="image" src="https://github.com/Textualize/textual/assets/49741340/059f5f61-16cc-43b6-9623-68e47748717f">

Reply:
I'd recommend reading the section on [Syntax Highlighting](https://textual.textualize.io/widgets/text_area/#syntax-highlighting) in the TextArea docs. The syntax highlighting won't work simply by including the highlights `scm` file, you  would need to map the `@symbol` names to a Rich style in the theme.